packages/trpc/server/routers/viewer/apps.tsx

```diff
@@ -0,0 +1,237 @@
+  type: string;
+  description: string;
+  keys: unknown;
+  enabled: boolean;
+}
+
+export const appsRouter = router({
+  listLocal: authedProcedure
+    .input(
+      z.object({
+        variant: z.string(),
+      })
+    )
+    .query(async ({ ctx, input }) => {
+      if (ctx.session.user.role !== "ADMIN")
+        throw new TRPCError({
+          code: "UNAUTHORIZED",
+        });
+
+      const localApps = getLocalAppMetadata();
+      const dbApps = await ctx.prisma.app.findMany({
+        where: {
+          categories: {
+            has: input.variant === "conferencing" ? "video" : (input.variant as AppCategories),
+          },
+        },
+        select: {
+          slug: true,
+          keys: true,
+          enabled: true,
+        },
+      });
+
+      const filteredApps: FilteredApp[] = [];
+
+      for (const app of localApps) {
+        if (app.variant === input.variant) {
+          // Find app metadata
+          const dbData = dbApps.find((dbApp) => dbApp.slug === app.slug);
+
+          // If the app already contains keys then return
+          if (dbData?.keys) {
+            filteredApps.push({
+              name: app.name,
+              slug: app.slug,
+              logo: app.logo,
+              title: app.title,
+              type: app.type,
+              description: app.description,
+              keys: dbData.keys,
+              enabled: dbData?.enabled || false,
+            });
+          } else {
+            const appKey = deriveAppDictKeyFromType(app.type, appKeysSchemas);
+            const keysSchema = appKeysSchemas[appKey as keyof typeof appKeysSchemas] || null;
+            filteredApps.push({
+              name: app.name,
+              slug: app.slug,
+              logo: app.logo,
+              type: app.type,
+              title: app.title,
+              description: app.description,
+              enabled: dbData?.enabled || false,
+              keys: keysSchema ? keysSchema.keyof()._def.values : null,
+            });
+          }
+        }
+      }
+      return filteredApps;
+    }),
+  toggle: authedProcedure
+    .input(
+      z.object({
+        slug: z.string(),
+        enabled: z.boolean(),
+      })
+    )
+    .mutation(async ({ ctx, input }) => {
+      if (ctx.session.user?.role !== "ADMIN")
+        throw new TRPCError({
+          code: "UNAUTHORIZED",
+        });
+
+      const { prisma } = ctx;
+
+      const app = await prisma.app.update({
+        where: {
+          slug: input.slug,
+        },
+        data: {
+          enabled: !input.enabled,
+        },
+      });
+
+      // Get app name from metadata
+      const localApps = getLocalAppMetadata();
+      const appMetadata = localApps.find((localApp) => localApp.slug === app.slug);
+
+      // If disabling an app then we need to alert users basesd on the app type
+      if (input.enabled) {
+        if (app.categories.some((category) => category === "calendar" || category === "video")) {
```

Comment:
We can't filter on event types with specific video apps because they are stored as an array of JSON objects and Prisma does not support it for Postgres yet.

https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filtering-on-object-key-value-inside-array